smithy-kotlin-codegen/build.gradle.kts

```diff
@@ -42,6 +42,8 @@ val generateSdkRuntimeVersion by tasks.registering {
     // this keeps us from having to manually change version numbers in multiple places
     val resourcesDir = "$buildDir/resources/main/software/amazon/smithy/kotlin/codegen/core"
     val versionFile = file("$resourcesDir/sdk-version.txt")
+    val gradlePropertiesFile = rootProject.file("gradle.properties")
+    inputs.file(gradlePropertiesFile)
```

Comment:
question

This is to determine build dirty-ness or something else?

Reply:
Yes. When we upgrade the version the generated version is not changing because it is already generated. This generally seems to only affect local development because CI is a fresh build. The fix is to clean and rebuild (or manually delete the file).